lib/java/src/org/apache/thrift/TServiceClient.java

```diff
@@ -85,7 +85,11 @@ protected void receiveBase(TBase<?,?> result, String methodName) throws TExcepti
       throw new TApplicationException(TApplicationException.BAD_SEQUENCE_ID,
           String.format("%s failed: out of sequence response: expected %d but got %d", methodName, seqid_, msg.seqid));
     }
-    result.read(iprot_);
+    try {
```

Comment:
This change is incorrect.  If read throws, the connection is closed.  It means the client sent part of a response but didn't get all of it out, and disconnected, or perhaps in your case, the type of a previously existing field was changed between releases, so the deserializer was looking for one field but got another.  In this case, the deserializer is first looking for a list length, then for each entry in the list it is looking for a string length and a string.  By changing field #2 to a string, it has corrupted the stream in an incompatible way.

Reply:
"If read throws, the connection is closed"——Have to say that is is not true and misleading, in most cases people use keep-alive mode in thrift and a connection is shared between multi requests, the connection is still active even if read throws an exception.